src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -1092,3 +1118,43 @@ function getSessionId(): string {
 
     return sessionId;
 }
+
+async function createClientPipeTransport(
+    pipeName: string,
+    encoding: RAL.MessageBufferEncoding = 'utf-8'
+): Promise<PipeTransport> {
+    const connected = new Promise<[MessageReader, MessageWriter]>((resolve) => {
+        const server: net.Server = net.createServer((socket: net.Socket) => {
+            // Sever has connected
+            server.close();
+            resolve([new SocketMessageReader(socket, encoding), new SocketMessageWriter(socket, encoding)]);
+        });
+
+        // Start the server listening on the pipe name.
+        const pipeConnectionString = getClientPipeName(pipeName);
+        server.listen(pipeConnectionString);
+    });
+
+    // Provide onConnected so that calling method can spin up the pipe client
+    return {
+        onConnected: async () => {
+            return await connected;
+        },
+    };
+}
+
+function getClientPipeName(pipeName: string): string {
+    const isWindows = os.platform() === 'win32';
+
+    if (isWindows) {
+        // Windows pipes are named using the following format
+        return '\\\\.\\pipe\\' + pipeName;
+    } else {
+        // Unix-type pipes are are actually writing to a file
+        return path.join(os.tmpdir(), pipeName + '.sock');
```

Comment:
as in the other PR, the pipename should include the tmpdir so we don't have to re-calculate it on the server.